source/adios2/core/IO.cpp

```diff
@@ -198,8 +198,8 @@ bool IO::RemoveAttribute(const std::string &name) noexcept
     if (itAttribute != m_Attributes.end())
     {
         // first remove the Variable object
-        const std::string type(itAttribute->second.first);
-        const unsigned int index(itAttribute->second.second);
+        const std::string type(itAttribute->second.m_Type);
+        const unsigned int index(itAttribute->second.m_Index);
 
```

Comment:
Picking this as an example.

Reply:
yeah, we are mapping to a local variable called `type` and `index`....seems subjective to me

Reply:
But it isn't really a pair, it just happens to contain two values... I agree it is subjective, as is this article: http://maintainablecode.logdown.com/posts/158531-stdpair-considered-harmful, but that's where I'm coming from.

Anyway, should I just redo this PR without the DataMapEntry stuff?

Reply:
@germasch nice article, but context matters.....in this case you had no problem understanding things when looking at the entire section, not a couple of lines. I'd rather switch GetType < T > to fixed width types so bpls can present types like numpy does (without the non-portable names).